saml-authentication-server/src/main/java/jetbrains/buildServer/auth/saml/plugin/SamlAuthenticationScheme.java

```diff
@@ -150,6 +154,16 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
                 return sendUnauthorizedRequest(request, response, String.format("SAML request NOT authenticated for user id %s: user with such username or %s property value not found", username, SamlPluginConstants.ID_USER_PROPERTY_KEY));
             }
 
+            if (settings.isAssignGroups()) {
+                String samlGroups = getAttribute(auth, settings.getGroupsAttributeMapping());
+                LOG.debug(String.format("SAML Groups = '%s'", samlGroups));
+
+                // Process the SAML groups assigned to this user
+                if (!samlGroups.isEmpty()) {
```

Comment:
Here you've lost a use case when somebody wants to completely remove user from all the groups - i.e. if I set "remove unassigned groups = True" I would expect that the groups will be cleared for the user, not just skipped. I would rather remove this condition and let the processGroups logic handle the rest (don't forget the null check of samlGroups then)

Reply:
Ah, good point... Will update.

Reply:
Addressed in [`8878dcc` (#52)](https://github.com/morincer/teamcity-plugin-saml/pull/52/commits/8878dccd3ef2513420554cdeb65d19de6d711d59)